### DISCUSSION TITLE
Review, vetting, and submission workflows

### DISCUSSION THREAD

**Discussion by michael-wynne-wsu** [MukurtuCMS/Mukurtu-CMS#34] [New Feature Request]
There have been user requests for a series of related features. Input is required to define these before we can explore development.

**Review workflows and tools**
- Allow contributors to create content which must be reviewed by a protocol steward before publishing.
- Is a communication/feedback system needed for back-and-forth editing?

**Submission workflows and tools**
- Allow anonymous or registered users to submit content (eg: digital heritage, dictionary word) without having a contributor user role.
- Through a structured submission form?
- How to contribute media?
- How to communicate 3Cs?

**Comment by jpringle1980** [comment]
In thinking about our unique needs, having a manner in which freshly-submitted content is placed in some sort of queue for associated approvers (i.e. protocol stewards) before it becomes publicly visible (to either anonymous or registered users) would be very helpful. I realize that we could simply create a new protocol (i.e. "material pending approval for access") and provide content creators with the necessary instructions to place all items under that protocol, but I would also be open to some sort of alert that can be made to protocol stewards once content has been submitted. Then, rather than new content being associated with the aforementioned temporary protocol, it could instead simply have some sort of approval (yes/no-connect with xyz protocol) radio button, much like switching a user account request from 'blocked' to 'active.' 

I had not considered any sort of back-and-forth editing process, as I assume that what is deposited in our Mukurtu-supported database would be the equivalent of a 'final' product. Though, I am curious about how Mukurtu could potentially be used to assist with collaboration of new content, live and dynamically. Sort of a peer-review layer? Maybe implement some sort of tracking/accountability tool for edits made to DH items? Not sure if this is currently available, but I could see it being helpful to run a report on any particular DH item and get a history of any edits made and by whom. 

I believe the existing submission fields for a new DH item are self-explanatory, so I don't think I see the need to create yet another, more simplified, submission form. Instead, I will focus on creating documentation for new content creators and provide any necessary training to familiarize them with the environment. But that's just me and others might feel differently.

**Reply by michael-wynne-wsu** [reply]
> having a manner in which freshly-submitted content is placed in some sort of queue for associated approvers (i.e. protocol stewards) before it becomes publicly visible (to either anonymous or registered users) would be very helpful

Would this look something like... protocol is selected when importing the item as usual, but the item is set to "unpublished" (or something, separate from protocols). Then a protocol steward would be notified of new content and have the ability to toggle from unpublished to published?

**Reply by jpringle1980** [reply]
Yeah! I think that would be a great solution.

**Reply by michael-wynne-wsu** [reply]
> Not sure if this is currently available, but I could see it being helpful to run a report on any particular DH item and get a history of any edits made and by whom.

Drupal admins *can* access versioning history, but it's not something we've had a need to expose more on the front end, will look into that.

**Reply by michael-wynne-wsu** [reply]
> Yeah! I think that would be a great solution.

So then follow-ups, who can do the following things?
1) When creating a record, set it as published (effectively bypassing the review step)? I would think a protocol steward.
2) Toggle a record between published and unpublished. I would think a protocol steward.
3) Can say a contributor ever have permission to publish without a protocol steward reviewing it? If so, how is that managed?
